src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -131,7 +131,7 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   protected def findEntry(key: A): Entry =
     findEntry0(key, index(elemHashCode(key)))
 
-  private[this] def findEntry0(key: A, h: Int): Entry = {
+  protected[this] final def findEntry0(key: A, h: Int): Entry = {
```

Comment:
@retronym I agree about the binary compatibility. I'll check MiMa for a test case. I'd be surprised if this isn't covered or doesn't work. Could indeed be a problem of the Scala build.